### PR TITLE
Add channels support to api

### PR DESCRIPTION
**Pull request by dvorobiov**
@deeplook Can you please do me a favour and merge this pull request into python-sdk? New functionality to add / delete channels is arriving ;) There will be a few more endpoints to add later, but these are essential and we use them in integration tests. 
I guess you also may wish to add a new Channel class can be potentially added to Python SDK for convenience.

### CODE REVIEW

**Comment by deeplook** [comment]
One can argue about the usage of singular/plural forms in method names. I've tried to use more intuitive names than those of the literal endpoints where possible, like ``delete_device`` instead of ``delete_devices`` or ``get_device_apps`` instead of ``get_devices_apps``. The singular form is more appropriate for method names (especially on instances of the respective resource class) and the plural form more for endpoint path names. I'd like to stick to that strategy, because it makes the API easier to learn and "forsee". So I'd prefer the method names ``post_channel``, ``delete_channel`` and ``delete_channels``.

Speaking of resource classes, I don't see corresponding methods in ``Device`` which would probably make most sense.

I'll make some more comments on individual lines.

And then, I prefer PRs with tests included.

File: relayr/api.py [hunk 1]
```diff
@@ -1037,6 +1037,54 @@ def get_device_apps(self, deviceID):
...
+        :type channelID: string
+        :rtype: no content
+        """
+        url = '{0}/channels/{1}'.format(self.host, channelID)
+        _, res = self.perform_request('DELETE', url, headers = self.headers)
+        return res
+
+    def delete_channels_batch(self, deviceID = None, transport = None):
+        """
+        Delete user's channels by criteria.
```
Comment (deeplook) [inline relayr/api.py]: Why *user's* channel? There is no reference to any user in this method's code.

File: relayr/api.py [hunk 2]
```diff
@@ -1037,6 +1037,54 @@ def get_device_apps(self, deviceID):
...
+                                      data=data, headers = self.headers)
+        return res
+
+    def delete_channels(self, channelID):
+        """
+        Delete an existing channel by its id.
+        
+        :param channelID: the UUID of the channel
+        :type channelID: string
+        :rtype: no content
```
Comment (deeplook) [inline relayr/api.py]: That's not a type.

File: relayr/api.py [hunk 3]
```diff
@@ -1037,6 +1037,54 @@ def get_device_apps(self, deviceID):
         _, data = self.perform_request('GET', url, headers=self.headers)
         return data
 
+    def post_channels(self, deviceID, transport):
+        """
+        Create a new channel.
```
Comment (deeplook) [inline relayr/api.py]: Too terse, a channel to what?

File: relayr/api.py [hunk 4]
```diff
@@ -1037,6 +1037,54 @@ def get_device_apps(self, deviceID):
...
+        
+        :param deviceID: the device UUID
+        :type deviceID: string
+        :param transport: transport for channel (mqtt, websockets, etc.)
```
Comment (deeplook) [inline relayr/api.py]: What are allowed "transports"? Wouldn't the term channel *type* or *kind* be clearer?

File: relayr/api.py (hunk 1 above)
Comment (dvorobiov) [inline relayr/api.py]: As soon as you have provide a token, which is basically the identity of the user  and the app. Moreover, when you delete device you can also delete other users' channels too. 
Regarding naming: I would follow whatever rules you set in your codebase. I checked the names above and below and they followed plurals like REST. I ll rename it.

File: relayr/api.py (hunk 2 above)
Comment (dvorobiov) [inline relayr/api.py]: Will remove.

File: relayr/api.py (hunk 3 above)
Comment (dvorobiov) [inline relayr/api.py]: OK

File: relayr/api.py (hunk 4 above)
Comment (dvorobiov) [inline relayr/api.py]: Explanations for  transports will be added to the official docs. Type and kind doesn't describe channels - they are too generic. We use transport everywhere within out system as it describes the transport used by channels to deliver messages.

**Comment by dvorobiov** [comment]
@deeplook  I updated PR with renaming. I generally agree with you on unit tests, but it takes time for me to come up with them in the language I am just familiar with. So I cannot do it now.

**Pull request merged**